### ISSUE TITLE
Filter out plant contamination from sketch file

### ISSUE THREAD

**Issue by ZarulHanifah** [sourmash-bio/sourmash#2668]
Hello sourmash developers,

I tried sourmash signature subtract to remove host plant from metagenome assembly sketch:

```
sourmash signature subtract \
    results/sketch_euks/D1B.sig results/oil_palm.sig
```

But I got this error message:
```
== This is sourmash version 4.6.1. ==
== Please cite Brown and Irber (2016), doi:10.21105/joss.00027. ==

Traceback (most recent call last):
  File "/fs03/ps45/Zarul/am/workflow/sour_fastq/.snakemake/conda/404cf1709c977b362ad46c932bc47d0b_/bin/sourmash", line 11, in <module>
    sys.exit(main())
  File "/fs03/ps45/Zarul/am/workflow/sour_fastq/.snakemake/conda/404cf1709c977b362ad46c932bc47d0b_/lib/python3.10/site-packages/sourmash/__main__.py", line 13, in main
    return mainmethod(args)
  File "/fs03/ps45/Zarul/am/workflow/sour_fastq/.snakemake/conda/404cf1709c977b362ad46c932bc47d0b_/lib/python3.10/site-packages/sourmash/cli/sig/subtract.py", line 54, in main
    return sourmash.sig.__main__.subtract(args)
  File "/fs03/ps45/Zarul/am/workflow/sour_fastq/.snakemake/conda/404cf1709c977b362ad46c932bc47d0b_/lib/python3.10/site-packages/sourmash/sig/__main__.py", line 571, in subtract
    from_sigobj = sourmash.load_one_signature(from_sigfile, ksize=args.ksize, select_moltype=moltype)
  File "/fs03/ps45/Zarul/am/workflow/sour_fastq/.snakemake/conda/404cf1709c977b362ad46c932bc47d0b_/lib/python3.10/site-packages/sourmash/signature.py", line 446, in load_one_signature
    raise ValueError("expected to load exactly one signature")
ValueError: expected to load exactly one signature
```

The D1B.sig I generated using command:
```
sourmash sketch dna D1B_assembly.fasta \
    -p k=21,k=31,k=51,abund \
    -o D1B.sig &> {log}
```

While host_plant.sig I generated using command:
```
sourmash sketch dna host_plant.fasta \
    -p k=21,k=31,k=51,abund \
    -o host_plant.sig &> {log}
```

Did I get this right?

Thank you.

**Comment by ctb** [comment]
hi @ZarulHanifah, sorry, you ran into something that ...ok it kinda sucks, but it's hard to imagine doing differently on our end at the moment, because I think that it would cause problems for less advanced users!

tl;dr you need to run the subtract for each ksize individually - specify `-k 21` and then `-k 31` and then `-k 51` separately.

Can you give at least one ksize a try and let me know that that works? Thanks!

**Comment by ZarulHanifah** [comment]
Works wonderfully! Thank you so much Titus!
